core/src/main/java/org/elasticsearch/script/ClassPermission.java

```diff
@@ -89,6 +89,7 @@
             java.lang.Object.class.getName(),
             java.lang.Short.class.getName(),
             java.lang.String.class.getName(),
+            java.lang.StringBuffer.class.getName(),
```

Comment:
It makes the error message when someone tries to use Groovy assertions point a bit more to Groovy instead of us (ie, NoClassDefFoundError for InvokerHelper rather than for StringBuffer), so I'm in favor of keeping it. If there is some security issue with it though, we can definitely leave it out. I'm fine either way.